From c9f944fa45d760dd2aeb1fd0d2187f9105e747df Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 26 Apr 2019 22:50:13 +0200
Subject: [PATCH 63/90] md: annotate mddev_trylock()

HANDLES:

drivers/md/md.c:8963:3: warning: releasing mutex 'mddev->reconfig_mutex' that was not held [-Wthread-safety-analysis]
                mddev_unlock(mddev);
                ^
drivers/md/md.c:9112:4: warning: releasing mutex 'mddev->reconfig_mutex' that was not held [-Wthread-safety-analysis]
                        mddev_unlock(mddev);
                        ^
---
 drivers/md/md.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/md.h b/drivers/md/md.h
index b7d42b7a5f0d..addbb010ff78 100644
--- a/drivers/md/md.h
+++ b/drivers/md/md.h
@@ -512,7 +512,7 @@ static inline void mddev_lock_nointr(struct mddev *mddev) __acquires_mutex(mddev
 	mutex_lock(&mddev->reconfig_mutex);
 }
 
-static inline int mddev_trylock(struct mddev *mddev)
+static inline int mddev_trylock(struct mddev *mddev) __try_acquires_mutex(1, mddev->reconfig_mutex)
 {
 	return mutex_trylock(&mddev->reconfig_mutex);
 }
-- 
2.17.1

